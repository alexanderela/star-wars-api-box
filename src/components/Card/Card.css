@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");

.Card {
  background: url('../../assets/swapi_rough_gray.jpg');
  border-radius: 50px;
  color: white;
  cursor: pointer;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 1.1rem;
  height: 20rem;
  letter-spacing: 0.1rem;
}

.Card:hover {
  box-shadow: #d9ffd9 0px 0px 15px 2px;
}

.Card-active {
	box-shadow: #d9ffd9 0px 0px 15px 2px;
}

.fav-btn-card-container {
	align-items: center;
	display: flex;
	justify-content: space-between;
	padding: 0.5rem 2rem;
}

.fav-btn {
	align-items: center;
	border-radius: 50%;
	box-shadow: ;
	cursor: pointer;
	display: flex;
	height: 3.5rem;
	justify-content: center;
	width: 3.5rem;
}

.fav-btn:hover {
 	border-radius: 50%;
  box-shadow: #d9ffd9 0px 0px 20px 2px;
}

.fav-btn:focus {
  outline: 0;
}


.fav-btn-inactive {
	background-color: #FFF;
	border-radius: 50%;
}

.fav-btn-active {
	background-color: #ffdf00;
	border-radius: 50%;
}

.fa-jedi {
	color: #000;
	font-size: 2rem;
}

.card-text {
	margin: 0.5rem 0.75rem;
}

.card-header {
  color: #f9d500;
 	font-weight: 600;
}
