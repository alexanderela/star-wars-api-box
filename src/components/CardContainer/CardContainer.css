.CardContainer {
  background: url('../../assets/SWAPI-background.jpg');
	border: 1px solid black;
  border-radius: 10px;
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: repeat( auto-fit, minmax(17rem, 1fr) );
  margin: 0 5rem 3rem;
}

@media (max-width: 720px) {
  .CardContainer {
   margin: 0 auto 3rem
  }
}