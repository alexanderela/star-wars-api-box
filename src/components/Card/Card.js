import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';

const Card = ({
  entry, toggleFavorites, favorites, isFavorite,
}) => {
  const cardProperties = entry.properties.map(property => (
    <p key={property.header} className="card-text">
      <span className="card-header">
        {property.header}
      </span>
      {property.text}
    </p>
  ));

  return (
    <div className="Card">
      <div className="fav-btn-card-container">
        <h3>{entry.name}</h3>
        <button
          type="submit"
          className={`fav-btn 
						${(isFavorite || favorites.includes(entry))
            ? 'fav-btn-active'
            : 'fav-btn-inactive'}`}
          onClick={() => toggleFavorites(entry)}
        >
          <i className="fas fa-jedi" />
        </button>
      </div>
      {cardProperties}
    </div>
  );
};

Card.propTypes = {
  entry: PropTypes.object.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired,
  isFavorite: PropTypes.bool.isRequired,
};

export default Card;
