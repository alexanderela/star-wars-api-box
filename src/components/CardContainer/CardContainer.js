import React from 'react';
import './CardContainer.css';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const CardContainer = ({ entries, toggleFavorites, favorites }) => {
  const cards = entries.map(entry => (
    <Card
      entry={entry}
      key={entry.name}
      toggleFavorites={toggleFavorites}
      favorites={favorites}
      isFavorite={entry.isFavorite}
    />
  ));

  return (
    <div className="CardContainer">
      { cards }
    </div>
  );
};

CardContainer.propTypes = {
  entries: PropTypes.array.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired,
};

export default CardContainer;
