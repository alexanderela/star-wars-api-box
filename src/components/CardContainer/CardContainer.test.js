/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import Card from '../Card/Card'
import { shallow } from 'enzyme';
import people from '../../mockData/mockPeople.js';

const mockVehicle = [{
          "name": "Sand Cralwer",
          "id": "Sand Cralwer",
          "isFavorite": false, 
          "type":"vehicles",
          "properties": [
						{"header": 'Model: ', "text": "Digger Crawler"},
						{"header": 'Class: ', "text": "wheeled"},
						{"header": 'Passengers: ', "text": "1"}
					] 
        }]

describe('CardContainer component', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<CardContainer entries={mockVehicle}/>);
		expect(wrapper).toMatchSnapshot();
	})

	it('renders each card', () => {
		const wrapper = shallow(<CardContainer entries={mockVehicle} />);
		expect(wrapper.find(Card).length).toEqual(1)
	})
})

