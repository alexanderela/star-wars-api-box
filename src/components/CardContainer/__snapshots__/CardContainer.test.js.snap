// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    entries={
      Array [
        Object {
          "id": "Sand Cralwer",
          "isFavorite": false,
          "name": "Sand Cralwer",
          "properties": Array [
            Object {
              "header": "Model: ",
              "text": "Digger Crawler",
            },
            Object {
              "header": "Class: ",
              "text": "wheeled",
            },
            Object {
              "header": "Passengers: ",
              "text": "1",
            },
          ],
          "type": "vehicles",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          entry={
            Object {
              "id": "Sand Cralwer",
              "isFavorite": false,
              "name": "Sand Cralwer",
              "properties": Array [
                Object {
                  "header": "Model: ",
                  "text": "Digger Crawler",
                },
                Object {
                  "header": "Class: ",
                  "text": "wheeled",
                },
                Object {
                  "header": "Passengers: ",
                  "text": "1",
                },
              ],
              "type": "vehicles",
            }
          }
          isFavorite={false}
        />,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Sand Cralwer",
        "nodeType": "function",
        "props": Object {
          "entry": Object {
            "id": "Sand Cralwer",
            "isFavorite": false,
            "name": "Sand Cralwer",
            "properties": Array [
              Object {
                "header": "Model: ",
                "text": "Digger Crawler",
              },
              Object {
                "header": "Class: ",
                "text": "wheeled",
              },
              Object {
                "header": "Passengers: ",
                "text": "1",
              },
            ],
            "type": "vehicles",
          },
          "favorites": undefined,
          "isFavorite": false,
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            entry={
              Object {
                "id": "Sand Cralwer",
                "isFavorite": false,
                "name": "Sand Cralwer",
                "properties": Array [
                  Object {
                    "header": "Model: ",
                    "text": "Digger Crawler",
                  },
                  Object {
                    "header": "Class: ",
                    "text": "wheeled",
                  },
                  Object {
                    "header": "Passengers: ",
                    "text": "1",
                  },
                ],
                "type": "vehicles",
              }
            }
            isFavorite={false}
          />,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Sand Cralwer",
          "nodeType": "function",
          "props": Object {
            "entry": Object {
              "id": "Sand Cralwer",
              "isFavorite": false,
              "name": "Sand Cralwer",
              "properties": Array [
                Object {
                  "header": "Model: ",
                  "text": "Digger Crawler",
                },
                Object {
                  "header": "Class: ",
                  "text": "wheeled",
                },
                Object {
                  "header": "Passengers: ",
                  "text": "1",
                },
              ],
              "type": "vehicles",
            },
            "favorites": undefined,
            "isFavorite": false,
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
